### DISCUSSION TITLE
Separate IPs for MME and SGW Support?

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#10253] [Q&A]
We are working with a customer on S1 based connections to AGW from an eNB that requires unique IP for MME vs. SGW. 

Can this be implemented with Magma AGW? 

I have looked in /etc/magma/spgw.yml, and I see I can pick an interface, but can I specify an secondary IP on a given interface? 

Thanks,
Joey

**Comment by jpad-freedomfi** [comment]
@wallyrb and @sudhikan do you guys know above?

**Reply by sudhikan** [reply]
so from what I know, the SGW IP is not something that is statically configured in the eNB, it is sent to the eNB during `s1SetupResponse`. 

+ @uri200 as I recall having seem an interface separation for S1-MME and S1-U in TerraVM docs. Thoughts?

**Reply by uri200** [reply]
Teravm uses 2 different interfaces for S1 MME and S1-U

**Reply by pshelar** [reply]
This should work, agw has config for `s1u_iface_name` in spgw.yml and `s1ap_iface_name` in mme.yml

**Comment by jpad-freedomfi** [comment]
It can definitely be static configured. Nokia and Huawei cells allow static configuration of primary, secondary, etc. for SGW IPs for S1-U separate from S1-MME as you've said. 

In fact, based on our customer input, Huawei requires static configuration, and requires S1-U IP(s) to be different from S1-MME IP(s).

**Reply by edaspb** [reply]
Hi @jpad-freedomfi, what enodeB does your customer use? We  successfully use regular bts3900 Huawei enodeb with magma. Our version does require S1-U and S1-MME but allows configure the same IP for them.

**Reply by jpad-freedomfi** [reply]
thanks for the reply @edaspb! Does this configuration cause any traffic routing issues for you?

Here is our situation. Single eNB connecting to 2 cores. Core A has c/u plane separation, MME and SGW have different IP in different subnets and VLANs. We are trying to connect second S1 to Core B with MME/SGW which share IP (magma AGW). The customer concern is that the common IP will cause U-plane traffic to start using the C-plane interface/vlan or visa versa. 

Also, I have requested eNB model. Will share when I learn it.

**Reply by edaspb** [reply]
I don't think it will cause any issue. In huawei conseption each s1 and sctp links are independent and configuration itself is very flexible. There is a parameter called 'Local IP' which the system uses as source ip for establishing connections. Configuring the specific one for magma will avoid any unpredictable routing.

**Comment by jpad-freedomfi** [comment]
@sudhikan or @uri200 had another thought. 
Since in cfg file it looks like SPGW only cares about interface, not IP, could we just assign 2 IPs to the interface, and SPGW will listen on both? So while AGW may not care there are 2, eNB can feel happy with C/U plane separate end points?

**Reply by sudhikan** [reply]
cc: @pshelar 

I am not aware of the underlying plumbing to say 'yay' or 'nay' with confidence :-)

**Reply by uri200** [reply]
Tht will not work. `s1u_iface_name` is the name of the interface the [config script](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fpython%2Fscripts%2Fgenerate_oai_config.py&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=472&start_col=17&end_row=472&end_col=17) will use to extract the IP. It only gets one IP

I am not sure of this but maybe you can use a bridge to forward several interfaces to  the main `s1u_iface_name` interface. Maybe @pshelar  could help to see if that may work

**Reply by pshelar** [reply]
> Tht will not work. `s1u_iface_name` is the name of the interface the [config script](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fpython%2Fscripts%2Fgenerate_oai_config.py&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=472&start_col=17&end_row=472&end_col=17) will use to extract the IP. It only gets one IP
> 
> I am not sure of this but maybe you can use a bridge to forward several interfaces to the main `s1u_iface_name` interface. Maybe @pshelar could help to see if that may work

bridging would make it complicated since it works at L2. so I would not recommend it.
